src/Engine/ProtoCore/Reflection/Mirror.cs

```diff
@@ -468,20 +469,28 @@ public IEnumerable<MethodMirror> GetOverloads(string methodName)
             /// <summary>
             /// Given a method name, return the matching list of 
             /// constructors or static methods on this type and its base types
+            /// Excludes hidden methods from base types.
             /// </summary>
             /// <param name="methodName"></param>
             /// <returns></returns>
             public IEnumerable<MethodMirror> GetOverloadsOnType(string methodName)
             {
                 List<MethodMirror> members = new List<MethodMirror>();
-                IEnumerable<ClassMirror> baseClasses = this.GetClassHierarchy();
+                members.AddRange(GetConstructors().Where(x => x.MethodName == methodName));
+                members.AddRange(GetFunctions().Where(x => x.IsStatic && x.MethodName == methodName));
+
+                // Filter out hidden functions/properties:
+                // Create a set of unique function and constructor descriptions.
+                // In this case we use ToString() to get the unique description of the members (func signature, constructor names).
+                var derivedClassMembers = new HashSet<string>();
+                members.ForEach(x => derivedClassMembers.Add(x.ToString()));
```

Comment:
Are you going to make the same simplification in this [commit ](fc44db9961993c85ab8af5cec4a8a2fe22ffaadd) over here?

Reply:
`GetOverloadsOnType` must show duplicate names, so I cannot apply the simplification.
The existing code will filter duplicate signatures .